### PR TITLE
Put modules of examples in separate files

### LINKED ISSUE #321: Split examples into mod/imp

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Fixes #321

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
you need to rebase your branch on top of master `git pull origin master --rebase` & then `git push origin examples -f`. Also always prefix your branches with your pseudo/name to avoid any collisions in the future.

**Comment by Hofer-Julian** [comment]
> you need to rebase your branch on top of master 

As far as I can tell, I am already rebased on top of master

> Also always prefix your branches with your pseudo/name to avoid any collisions in the future.

Sound like a good system, will give my best to think of it for my next PRs :)

File: examples/builder_pattern/main.rs [hunk 1]
```diff
@@ -1,6 +1,13 @@
 use gtk::prelude::*;
 use gtk::{Align, Application, ApplicationWindowBuilder, ButtonBuilder};
 
+fn main() {
+    let application =
+        Application::new(Some("com.github.gtk-rs.examples.basic"), Default::default());
```
Comment (bilelmoussaoui) [inline examples/builder_pattern/main.rs:6]:
```suggestion
        Application::new(Some("com.github.gtk-rs.examples.builder_pattern"), Default::default());
```
it's a bit unrelated, but let's use a different app-id per example if possible

File: examples/composite_template/ex_application_window/imp.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    #[template_child]
+    pub headerbar: TemplateChild<gtk::HeaderBar>,
+    #[template_child]
+    pub label: TemplateChild<gtk::Label>,
+    // You can specify the optional `id` argument if the id is not the same
+    // as the identifier
+    #[template_child(id = "subtitle_label")]
+    pub subtitle: TemplateChild<gtk::Label>,
+    #[template_child]
+    pub menubutton: TemplateChild<crate::ex_menu_button::ExMenuButton>,
```
Comment (bilelmoussaoui) [inline examples/composite_template/ex_application_window/imp.rs:20]:
```suggestion
    pub menubutton: TemplateChild<ExMenuButton>,
```
& use `use crate::ex_menu_button::ExMenuButton`; maybe?

File: examples/custom_editable/main.rs [hunk 3]
```diff
@@ -8,35 +8,19 @@ use gtk::prelude::*;
...
+        Some("com.github.gtk-rs.examples.editable"),
+        Default::default(),
+    );
+
+    application.connect_activate(build_ui);
+    application.run();
+}
 
 fn build_ui(application: &gtk::Application) {
     let provider = gtk::CssProvider::new();
```
Comment (bilelmoussaoui) [inline examples/custom_editable/main.rs:22]: Maybe we could be a bit more correct here and setup the css on connect_startup :-)

File: examples/custom_paintable/custom_paintable/imp.rs [hunk 4]
```diff
@@ -0,0 +1,81 @@
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib, graphene, gsk};
+#[derive(Default)]
```
Comment (bilelmoussaoui) [inline examples/custom_paintable/custom_paintable/imp.rs:4]: Empty line between after the imports
```suggestion

#[derive(Default)]
```

**Comment by bilelmoussaoui** [comment]
Looks good to me otherwise, thanks!

File: examples/composite_template/ex_application_window/imp.rs (hunk 2 above)
Comment (Hofer-Julian) [reply examples/composite_template/ex_application_window/imp.rs]: Fixed

File: examples/custom_editable/main.rs (hunk 3 above)
Comment (Hofer-Julian) [reply examples/custom_editable/main.rs]: Fixed

**Comment by Hofer-Julian** [comment]
Should be fine now

File: examples/custom_layout_manager/custom_layout_child/imp.rs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib, graphene};
+use once_cell::sync::Lazy;
+use std::cell::RefCell;
+
+#[derive(Debug)]
+pub struct CustomLayoutChild {
+    pub color: RefCell<gdk::RGBA>,
+}
+impl Default for CustomLayoutChild {
```
Comment (bilelmoussaoui) [inline examples/custom_layout_manager/custom_layout_child/imp.rs:12]: empty line :)

File: examples/composite_template/main.rs [hunk 6]
```diff
@@ -1,143 +1,8 @@
...
-            .subtitle
-            .set_text("This is an example window made using composite templates");
-    }
-}
+mod ex_application_window;
+pub mod ex_menu_button;
 
+use ex_application_window::ExApplicationWindow;
+use gtk::prelude::*;
 fn main() {
```
Comment (bilelmoussaoui) [inline examples/composite_template/main.rs:6]: empty line

**Review by bilelmoussaoui — APPROVED** [review]
just two little missing new lines nitpicks :)

**Comment by bilelmoussaoui** [comment]
Btw, another example got merged recently, would be nice if you could rebase on top of master & update that one as well so that all the examples are consistent

**Comment by sdroege** [comment]
Thanks for the cleanup :)

**Comment by Hofer-Julian** [comment]
> Btw, another example got merged recently, would be nice if you could rebase on top of master & update that one as well so that all the examples are consistent

Done

**Review by bilelmoussaoui — APPROVED** [review]

**Comment by bilelmoussaoui** [comment]
Thanks Julian :)

**Pull request merged**